### ISSUE TITLE
Add support for Content-Type:"application/json; charset=UTF-8" style header

### ISSUE THREAD

**Issue by naufraghi** [timothycrosley/hug#123]
I'm having some problem in a toy app, I'm experiencing a server error in Firefox but not in Chrome (same client).

Diving in the code I noticed the problem is related with the dictionary lookup in [HugAPI.input_format](https://github.com/timothycrosley/hug/blob/develop/hug/decorators.py#L67).
```python
return getattr(self, '_input_format', {}).get(content_type, hug.defaults.input_format.get(content_type, None))
```

Whenever the content type is composed nothing matches.

An easy workaround is to strip after the `;` but generally a real fix requires an encoding argument floating around, with an `UTF-8` default.

I'd like to help, you'd like to discuss an approach to solve the problem?

**Comment by timothycrosley** [comment]
Hi @naufraghi,

Thanks for reporting this issue! I think the ideal solution would be to update the input_handlers here: https://github.com/timothycrosley/hug/blob/develop/hug/input_format.py to take an optional encoding that defaults to utf8, and then update the code that checks for input handlers here https://github.com/timothycrosley/hug/blob/develop/hug/decorators.py#L337 to separate on ';' checking for the encoding hint and passing it to the input handlers if it is found.

Let me know if this makes sense.

Thanks!

~Timothy

**Comment by naufraghi** [comment]
Fix almost done, working on the tests.

**Comment by timothycrosley** [comment]
@naufraghi Great! Good to hear! Let me know if you need any help writing the tests, or otherwise.

Thanks!

~Timothy

**Comment by naufraghi** [comment]
Feature merged in #124
